### DISCUSSION TITLE
Connecting through SSH

### DISCUSSION THREAD

**Discussion by N1GHTR4NG3R** [prisma/prisma#6476] [Q&A] [answered]
Hi, First off I will start with informing you I have never connected through an SSH Tunnel using Javascript, so forgive me if this comes across a little dumb...

I have installed `node-ssh` with NPM, and am trying to connect too my database with Prisma, by going through the SSH Tunnel...
I may have missed some steps out, I am not sure as I cannot find any "How to's" about connecting this way, However this is what I have tried, from another thread found here: #2098

In the `.env` file I have modified the `DATABASE_URL` to display like so:

```
DATABASE_URL="mysql://ssh -N -L serverPort:user:databasePort localalhost@serverIP -i ./path/to/key"
```

However I get an error stating this is an invalid database string, and refers me to the docs page for database strings, which mentions nothing about connecting through SSH.

Would really appreciate it if someone could explain how I am meant to do this please, as I don't wish to open the DB up to the outside world.

Thanks in advance.

**Comment by ryands17** [comment, marked as answer]
@Tricky-Ricky 👋 

I have written an [article](https://dev.to/prisma/prisma-migrate-with-aws-aurora-serverless-53g7) on the same that you can follow to connect to a remote database via SSH tunnelling.

**Reply by N1GHTR4NG3R** [reply]
> I have written an [article](https://dev.to/prisma/prisma-migrate-with-aws-aurora-serverless-53g7) on the same that you can follow to connect to a remote database via SSH tunnelling.

Funnily enough I did look at that before coming here, through a google search, Unfortunately it didn't make much sense to me, as I don't use AWS. But much appreciated.

I did however figure out what I had to do, and was a lot simpler than I first thought, so I'll explain here just in-case anyone else searches.

Installed `node-ssh` through npm.
On my `dbConnection.js` file I did this:
```ts
import { createRequire } from "module";
const require = createRequire(import.meta.url);

import Prisma from "@prisma/client";
const { PrismaClient } = Prisma;
const db = new PrismaClient();

const { NodeSSH } = require("node-ssh");
const ssh = new NodeSSH();

ssh
  .connect({
    host: "ipAddress",
    port: "portForServer",
    username: "serverUser",
    privateKey: "Relative/path/for/privateKey",
    tryKeyboard: true,
  })
  .then(() => {
    console.log("Connected through SSH, attempting Database connection...");
    if (db) {
      console.log("Database connected successfully!");
    } else {
      console.log("Failed to connect to database...");
    }
  });
  ```
  Then upon starting my server, It will initiate the SSH tunnel and connect, then initiate the Database connection using the Prisma URL, which remained unchanged except using localhost, instead of a direct IP.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
